transport/messages/result_message_base.hh

```diff
@@ -49,14 +49,15 @@ public:
     void add_tablet_info(locator::tablet_replica_set tablet_replicas, std::pair<dht::token, dht::token> token_range) {
         if (!tablet_replicas.empty()) {
             auto replicas_values = make_list_value(replica::get_replica_set_type(), replica::replicas_to_data_value(tablet_replicas));
-            this->add_custom_payload("tablet_replicas", replicas_values.serialize_nonnull());
             auto v1 = data_value(dht::token::to_int64(token_range.first));
             auto v2 = data_value(dht::token::to_int64(token_range.second));
-            bytes token_bytes(bytes::initialized_later(), v1.serialized_size() + v2.serialized_size());
-            auto i = token_bytes.begin();
+
+            bytes tablet_routing_bytes(bytes::initialized_later(), v1.serialized_size() + v2.serialized_size() + replicas_values.serialized_size());
+            auto i = tablet_routing_bytes.begin();
             v1.serialize(i);
             v2.serialize(i);
```

Comment:
These are not CQL values. They are native protocol values, so using CQL data value serialization is wrong, even if it happens to lead to the same result (I think it doesn't for the list).

See the primitives in transport::response::write_*(). You can't use them directly, but write_long() isn't too complicated.

Reply:
Not sure if I understand what exactly the problem is. Should I not use the `serialize()` method?

Reply:
@avikivity What's wrong with saying that these are CQL values? The RESPONSE message contains CQL values, and every driver should know how to deserialize them.

The advantage of using CQL values is that the value will match what a query of system.tablets would return, so there is nice symmetry which could be leveraged.

Reply:
We could say it's a CQL value. But then we need to define it using CQL terms, not [int][short][long] etc.

I don't think there's precedent for sending CQL values in the native protocol, apart from a ResultSet.

Reply:
So would using notation like the following at the beginning be appropriate in that case? Something like:  `LongType LongType ListType(TupleType(UUIDType, Int32Type))`.

Reply:
We shouldn't mix styles. If we pick CQL value, then it's TupleType(LongType, LongType, ListType(TupleType(UUIDType, Int32Type)))))))))))))))))))))))))))))) - one value, not an ad-hoc tuple.

And carefully documented the meaning of each field. It should be possible to write a driver without referring to the source code.